gagbo-truly-cross-platform.md

```diff
@@ -0,0 +1,137 @@
+Make your setup truly cross-platform
+===============================================================================
+Once you have taken the time to setup Vim exactly the way you want, you might
+still encounter configuration issues. My whole `~/.vim` folder is under version
+control so I can just clone it on any computer I want my familiar Vim
+experience.
+
+But I use Windows computers and Linux computers. Even within the
+same OS, I might have different dependencies installed for the plugins I try to
+import. Also, from time to time I like to use Neovim too, to try out a few
+unique features.
+
+
+
+OS specific settings
+-------------------------------------------------------------------------------
+To make OS specific settings we need to know the OS we are
+running on.
+We can ask vim directly if it was compiled for Windows, otherwise a system call
+to `uname` will give the running OS. The function below returns a string
+containing the value of the running OS.
+
+```vimscript
+function! whichEnv() abort
+    if has('win64') || has('win32') || has('win16')
+        return 'WINDOWS'
+    else
+       return toupper(substitute(system('uname'), '\n', '', ''))
+    endif
+endfunction
+
+" Later use in another file
+if (whichEnv() ==# 'WINDOWS')
+    " Enable Windows specific settings/plugins
+else if (whichEnv() ==# 'LINUX')
+    " Enable Linux specific settings/plugins
+else if (whichEnv() ==# 'DARWIN')
+    " Enable MacOS specific settings/plugins
+else
+    " Probably an error
+endif
+```
+
+Intermission : system calls optimizations
+-------------------------------------------------------------------------------
+System calls are costly (see the difference in startuptime with or without the
+no-stl-wednesdays plugin which is only a call to `date`); caching the results as
```

Comment:
The reference to no-statusline-wednesday made me chuckle but at this point it's just an in-joke among a handful of #vim regulars so I'm not sure it's valuable in the current context.

Reply:
I agree, this paren should have been a quote to follow the little convention I had ; read this as 

> note to myself : pull and show the code from no-stl-wednesday -> run vim --startuptime to show how innocent code can have very strong effects on startup time. Just because it has 'system' in it. So please use system carefully in your startup scripts